surefire-providers/common-junit48/src/main/java/org/apache/maven/surefire/common/junit48/GroupMatcherCategoryFilter.java

```diff
@@ -173,6 +176,11 @@ private boolean shouldRun( Description description, Description parent, Class<?>
         }
     }
 
+    private static boolean categoryIsInherited()
```

Comment:
IMO this `boolean` value never change, so it makes sense to cache it in the `private static final` field. It is also important due to the method `isAnnotationPresent` is implemented as calling reflection and creation of a new instance. This might be visible execution time in loops.